prov/rxm/src/rxm_ep.c

```diff
@@ -2040,92 +2040,6 @@ static int rxm_ep_bind(struct fid *ep_fid, struct fid *bfid, uint64_t flags)
 	return ret;
 }
 
-static int rxm_ep_ctrl(struct fid *fid, int command, void *arg)
-{
-	struct rxm_ep *rxm_ep;
-	int ret;
-
-	rxm_ep = container_of(fid, struct rxm_ep, util_ep.ep_fid.fid);
-
-	switch (command) {
-	case FI_ENABLE:
-		if (!rxm_ep->util_ep.rx_cq || !rxm_ep->util_ep.tx_cq)
-			return -FI_ENOCQ;
-		if (!rxm_ep->util_ep.av || !rxm_ep->cmap)
-			return -FI_EOPBADSTATE;
-
-		/* At the time of enabling endpoint, FI_OPT_BUFFERED_MIN,
-		 * FI_OPT_BUFFERED_LIMIT should have been frozen so we can
-		 * create the rendezvous protocol message pool with the right
-		 * size */
-		ret = rxm_ep_txrx_pool_create(rxm_ep);
-		if (ret)
-			return ret;
-
```

Comment:
Was there an issue with old code? The pools are created after all relevant values have been set/modified by user.

Reply:
It also there, but Tx/Rx/RMA buffer pool allocation were moved to `rxm_ep_txrx_res_open`.
Also `rxm_ep_txrx_res_open` does initialization of Inject/Eager/SAR limits and sets min/max values for Multi-Recv and Buffered features. It is necessary, because Eager size depends on Buffered min value when user sets `FI_OFI_RXM_BUFFER_SIZE` env. variable to very small value (<= inject or <= (sizeof(RNDV hdr) + buffered_min))

I just want to have Tx/Rx/RMA pool and TAGGED/MSG Receive queus allocation in the same place - fi_enable is a good place for them.